### PR TITLE
Set Error.Path for HTTP exceptions

### LINKED ISSUE #1112: Rewriting remote error paths
Just wanted to start a discussion about remote error paths during schema stitching.

Currently if the server doesn't return a path on the error then `DelegateToRemoteSchemaMiddleware` doesn't add a path to the error object. This means for things like HTTP errors or other errors from the remote schema then on the stitched schema we don't have any error path to make diagnostic decisions.

Would it make sense to simply add the `context.Path` as the `error.Path` when there is no remote error?

### PR DESCRIPTION
**Pull request by nigel-sampson**
- Sets the Error.Path on HTTP exceptions for better error diagnostics.
- Add the schema name to the error extensions.

Resolves #1112

### CODE REVIEW

**Comment by michaelstaib** [comment]
There seems to be a local path captured in one of the snapshot tests ... you can do an ignore on that.

```
 Snapshot.Match<Person>(testPerson, matchOptions => matchOptions.IgnoreField("Size"))
```

File: src/Stitching/Stitching.Tests/Middleware/__snapshots__/DelegateToRemoteSchemaMiddlewareTests.HttpErrorsHavePathSet.snap [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "customer"
+      ],
+      "Locations": [
+        {
+          "Line": 3,
+          "Column": 29
+        }
+      ],
+      "Exception": {
+        "StackTrace": "   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()\n   at HotChocolate.Stitching.Utilities.HttpQueryClient.FetchInternalAsync(HttpQueryRequest request, HttpClient httpClient) in C:\\Dev\\GitHub\\hotchocolate\\src\\Stitching\\Stitching\\Utilities\\HttpQueryClient.cs:line 139\n   at HotChocolate.Stitching.Utilities.HttpQueryClient.FetchAsync(HttpQueryRequest request, HttpClient httpClient, IEnumerable`1 interceptors, CancellationToken cancellationToken) in C:\\Dev\\GitHub\\hotchocolate\\src\\Stitching\\Stitching\\Utilities\\HttpQueryClient.cs:line 46\n   at HotChocolate.Stitching.Delegation.RemoteQueryMiddleware.InvokeAsync(IQueryContext context) in C:\\Dev\\GitHub\\hotchocolate\\src\\Stitching\\Stitching\\Delegation\\RemoteQueryMiddleware.cs:line 39",
```
Comment (michaelstaib) [inline src/Stitching/Stitching.Tests/Middleware/__snapshots__/DelegateToRemoteSchemaMiddlewareTests.HttpErrorsHavePathSet.snap]: This path has to be ignored by the snapshot compare ... 

```csharp
 result.MatchSnapshot(matchOptions => matchOptions.IgnoreField("Errors[1].Exception.StackTrace"));
```

**Comment by michaelstaib** [comment]
@nigel-sampson looks good so far. We have just to fix the snapshot... I have left you a note. If you fix this the builds should go green.

**Comment by michaelstaib** [comment]
@nigel-sampson one other thing ... do you want it in for the 10.2.0 that we target for next week? If yes you have to merge it into `version_10_0_0_master`.

**Comment by nigel-sampson** [comment]
Editing the PR didn't have the right effect, will force push the PR soon.

**Comment by michaelstaib** [comment]
I really got issues with coverlet at the moment ... most of the build errors are due to a coverlet issue. Have to look into it. I rescheduled the v10 build .... should go green.

**Comment by michaelstaib** [comment]
I will merge as soon as the v10 is green.

**Comment by michaelstaib** [comment]
I will force this one through ... the coverlet issue is really a pain.

**Pull request merged**